src/views/pages/Login/index.js

```diff
@@ -43,6 +46,14 @@ class LoginPage extends Component {
         this.setState({ password: this.refs.password.value });
     };
 
+    handleLoginResponse = () => {
+        if (this.props.authentication.status === 'ERROR') {
+            if (!this.state.loginFailed) {
```

Comment:
This condition is confusing... Here we check if not loginFailed, but then set it to true...

Reply:
What if that state property was renamed from 'loginFailed' to 'loginFailedMessageDisplayed'? Would that make it clearer? I think the check was put there because if it is already set to 'true' we don't need to set it to true again.

Reply:
Oh that’s what this is doing. That’s fine it can also be ignored. 

I think in the general realm of React/Redux it’d be more common to see the reducer status field toggling message visibility. You essentially have a local component state field keying of global state. You’ve mirrored global state locally. If you just take out local state and toggle visibility with global state, you’ve acheived a cleaner more scalable pattern.

Reply:
That kind of makes sense to me, but I'm a bit confused. Should I be using the status that's set by the reducer then to toggle message visibility?

Reply:
I just think I'm more used to something like this:

```
render() {
    {...}
    {this.props.authentication.status === ERROR 
        ? (
            <ErrorNotice msg={this.props.authentication.message /> 
        ) : (
           null
        )
    } 
    {...}
}
```

Reply:
See how now matter what the redux action state was the component will do the right thing. No need for setState, we already rerendered because this action changed component props. Setting state would cause a second rerender. See what I'm getting at?

Reply:
Additionally, this automagical toggling will solve your problem above with the persistence layer.

Reply:
Okay, I can do that! Is there any way to reset those errors? I noticed that if you fail to create a user, then go to the login view, it shows the same error, because the props are the same.

Reply:
It'd be a big round about process. Honestly all these problems are solved by Redux Form. It keeps track of when fields are touched and resetting validation errors all that good stuff. Then you can have logic like, if the input has been touched and the state is error, display error message. For now I'd say ignore it and call it tech debt